static/js/stream_data.js

```diff
@@ -18,8 +18,16 @@ exports.clear_subscriptions = function () {
 
 exports.clear_subscriptions();
 
-exports.set_filter_out_inactives = function (want_filter) {
-    filter_out_inactives = want_filter || false;
+exports.set_filter_out_inactives = function () {
+    var has_many_streams;
+    if (page_params.demote_inactive_streams === 1) {
+        has_many_streams = exports.subscribed_subs().length >= 30;
+        filter_out_inactives = has_many_streams;
```

Comment:
The variable `has_many_streams` is not needed here, you can directly use `filter_out_inactives = exports.subscribed_subs().length >= 30;`